inference-engine/src/mkldnn_plugin/nodes/mkldnn_broadcast_node.cpp

```diff
@@ -0,0 +1,225 @@
+        broadcastType = broadcastNode->get_broadcast_spec().m_type;
+    }
+
+    auto * broadcastLayer = dynamic_cast<BroadcastLayer*>(getCnnLayer().get());
+    if (broadcastLayer == nullptr)
+        THROW_IE_EXCEPTION << "Cannot convert broadcast layer.";
+
+    if ((broadcast && getParentEdges().size() != 3) || (!broadcast && getParentEdges().size() != 2))
+        THROW_IE_EXCEPTION << "Incorrect number of input edges for layer " << getName();
+    if (!getChildEdges().size())
+        THROW_IE_EXCEPTION << "Incorrect number of output edges for layer " << getName();
+    if (getParentEdgeAt(0)->getDims().ndims() > getChildEdgeAt(0)->getDims().ndims())
+        THROW_IE_EXCEPTION << "Incorrect shapes for layer " << getName() << ". Input shape cannot be more than output shape";
+
+    if (broadcast) {
+        auto targetShapeBlob = dynamic_cast<TBlob<int> *>(getParentEdgeAt(1)->getParent()->getCnnLayer()->blobs["custom"].get());
+        if (!targetShapeBlob) {
+            THROW_IE_EXCEPTION << "Cannot get targetShapeBlob for layer " << getName();
+        }
+
+        auto ndims = getChildEdgeAt(0)->getDims().ndims();
+        auto srcDims = getParentEdgeAt(0)->getDims().ToSizeVector();
+        auto dstDims = getChildEdgeAt(0)->getDims().ToSizeVector();
+
+        if (broadcastType == ngraph::op::AutoBroadcastType::NUMPY) {
+            repeats = dstDims;
+            for (int i = 0; i < srcDims.size(); i++) {
+                repeats[ndims - 1 - i] /= srcDims[srcDims.size() - 1 - i];
+            }
+        } else if (broadcastType == ngraph::op::AutoBroadcastType::EXPLICIT) {
+            auto axesMappingBlob = dynamic_cast<TBlob<int> *>(getParentEdgeAt(2)->getParent()->getCnnLayer()->blobs["custom"].get());
+            if (!axesMappingBlob) {
+                THROW_IE_EXCEPTION << "Cannot get axesMappingBlob for layer " << getName();
+            }
+            auto axesMappingData = axesMappingBlob->buffer().as<int *>();
+
+            repeats = dstDims;
+            for (int i = 0; i < getParentEdgeAt(2)->getDims()[0]; i++) {
+                repeats[axesMappingData[i]] /= srcDims[i];
+                axesMapping.push_back(axesMappingData[i]);
+            }
+        } else {
+            THROW_IE_EXCEPTION << "Unsupported broadcast type for layer " << getName();
+        }
+
+        optimizedCase = isOptimizedCase(repeats);
+    }
+}
+
+void MKLDNNBroadcastNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    if (broadcast) {
+        supportedPrimitiveDescriptors = getSupportedConfigs(this, optimizedCase, repeats);
+    } else {
+        LayerConfig config;
+        DataConfig dataConfig, shapeConfig;
+        Precision dataPrecision = getCnnLayer()->outData[0]->getTensorDesc().getPrecision();
+        const SizeVector& data_dims = getCnnLayer()->insData[0].lock()->getTensorDesc().getDims();
```

Comment:
insData[0] is weak_ptr and must be validated

Reply:
fixed, added check